google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -266,19 +266,12 @@ def __init__(
 
         self._auth_init(credentials)
 
-        async def _async_init() -> None:
-            """Initialize InstanceConnectionManager's variables that require the
-            event loop running in background thread.
-            """
-            self._refresh_rate_limiter = AsyncRateLimiter(
-                max_capacity=2, rate=1 / 30, loop=self._loop
-            )
-            self._refresh_in_progress = asyncio.locks.Event()
-            self._current = self._loop.create_task(self._get_instance_data())
-            self._next = self._loop.create_task(self._schedule_refresh())
-
-        init_future = asyncio.run_coroutine_threadsafe(_async_init(), self._loop)
-        init_future.result()
+        self._refresh_rate_limiter = AsyncRateLimiter(
+            max_capacity=2, rate=1 / 30, loop=self._loop
+        )
+        self._refresh_in_progress = asyncio.locks.Event()
+        self._current = self._loop.create_task(self._get_instance_data())
+        self._next = self._loop.create_task(self._schedule_refresh())
```

Comment:
I think we already talked about this, but I think schedule_refresh should return a task directly.